src/diffusers/pipelines/spectrogram_diffusion/pipeline_spectrogram_diffusion.py

```diff
@@ -0,0 +1,75 @@
+import math
+from typing import Optional
+
+import torch
+
+from ...models.t5_attention import ContinuousContextTransformer
+from ...pipeline_utils import DiffusionPipeline, MelPipelineOutput
+from ...schedulers import DDPMScheduler
+
+
+class SpectrogramDiffusionPipeline(DiffusionPipeline):
+    def __init__(self, cont_context_trans: ContinuousContextTransformer, scheduler: DDPMScheduler) -> None:
```

Comment:
```suggestion
    def __init__(self, notes_encoder: SpectrogramNotesEnocder, continuous_encoder: SpectrogramContEncoder, decoder: T5FilmDecoder, scheduler: DDPMScheduler) -> None:
```

Reply:
If possible could we split the three different models as shown in the diagram here: 
![68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f6d757369632d73796e7468657369732d776974682d7370656374726f6772616d2d646966667573696f6e2f6172636869746563747572652e706e67](https://user-images.githubusercontent.com/23423619/200548760-d216cc33-41f4-435e-b8ff-0cc2ccfe1d7d.png)

into different classes? I think this makes the code a bit more agile and is easier to understand as well?

Reply:
Also I think we should pass the mel-to-sound model here as well so that people can directly generate music with the pipeline. IIRC the open-source implementation of the mel-to-sound model was not super easy (only coded up in TF) or is there a nice PT version that matches 1-to-1? Let me know if you need help :-)

Reply:
this pipeline class still needs to changed. Note that every independent components of the diagram should be it's own model as stated here: https://github.com/huggingface/diffusers/pull/1044/files#r1016473392